source/utilities.cc

```diff
@@ -0,0 +1,746 @@
+                                       const unsigned int components)
+    {
+      AssertThrow ((dynamic_cast<const GeometryModel::SphericalShell<dim>*> (&this->get_geometry_model()))
+          || (dynamic_cast<const GeometryModel::Box<dim>*> (&this->get_geometry_model())) != 0,
+          ExcMessage ("This ascii data plugin can only be used when using "
+                      "a spherical shell or box geometry."));
+
+
+      for (typename std::set<types::boundary_id>::const_iterator
+           boundary_id = boundary_ids.begin();
+           boundary_id != boundary_ids.end(); ++boundary_id)
+        {
+          const std_cxx11::array<unsigned int,dim-1> dimensions = get_boundary_dimensions(*boundary_id);
+
+          std_cxx11::shared_ptr<Utilities::AsciiDataLookup<dim-1> > lookup;
+          lookup.reset(new Utilities::AsciiDataLookup<dim-1> (components,
+                                                              Utilities::AsciiDataBase<dim>::scale_factor));
+          lookups.insert(std::make_pair(*boundary_id,lookup));
+
+          lookup.reset(new Utilities::AsciiDataLookup<dim-1> (components,
+                                                              Utilities::AsciiDataBase<dim>::scale_factor));
+          old_lookups.insert(std::make_pair(*boundary_id,lookup));
+
+
+          // Set the first file number and load the first files
+          current_file_number = first_data_file_number;
+
+          const int next_file_number =
+            (decreasing_file_order) ?
+            current_file_number - 1
+            :
+            current_file_number + 1;
+
+          this->get_pcout() << std::endl << "   Loading Ascii data boundary file "
+                            << create_filename (current_file_number,*boundary_id) << "." << std::endl << std::endl;
+          lookups.find(*boundary_id)->second->load_file(create_filename (current_file_number,*boundary_id));
+
+          // If the boundary condition is constant, switch
+          // off time_dependence immediately. This also sets time_weight to 1.0.
+          // If not, also load the second file for interpolation.
+          if (create_filename (current_file_number,*boundary_id) == create_filename (current_file_number+1,*boundary_id))
+            {
+              lookups.find(*boundary_id)->second.swap(old_lookups.find(*boundary_id)->second);
+              end_time_dependence (current_file_number, *boundary_id);
+            }
+          else
+            {
+              try
+                {
+                  this->get_pcout() << std::endl << "   Loading Ascii data boundary file "
+                                    << create_filename (next_file_number,*boundary_id) << "." << std::endl << std::endl;
+                  lookups.find(*boundary_id)->second.swap(old_lookups.find(*boundary_id)->second);
+                  lookups.find(*boundary_id)->second->load_file(create_filename (next_file_number,*boundary_id));
+                }
+              catch (...)
```

Comment:
I think you want to catch the very specific exception that is thrown by one of the functions above. If, for some reason, you get an exception of type `std::bad_alloc` because you're out of memory, then you don't want to just silently ignore this.